packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1969,28 +1969,87 @@ abstract class BuildContext {
+  /// (e.g. gesture callbacks) or timers, to obtain a value once, if that value
+  /// is not going to be cached and reused later.
+  ///
+  /// Calling this method is O(1) with a small constant factor, but will lead to
+  /// the widget being rebuilt more often.
+  ///
+  /// Once a widget registers a dependency on a particular type by calling this
+  /// method, it will be rebuilt, and [State.didChangeDependencies] will be
+  /// called, whenever changes occur relating to that widget until the next time
+  /// the widget or one of its ancestors is moved (for example, because an
+  /// ancestor is added or removed).
+  ///
+  /// The [aspect] parameter is only used when [T] is an
+  /// [InheritedWidget] subclasses that supports partial updates, like
+  /// [InheritedModel]. It specifies what "aspect" of the inherited
+  /// widget this context depends on.
+  T dependOnInheritedWidgetOfExactType<T extends InheritedWidget>({ Object aspect });
+
   /// Obtains the element corresponding to the nearest widget of the given type,
   /// which must be the type of a concrete [InheritedWidget] subclass.
   ///
   /// Calling this method is O(1) with a small constant factor.
   ///
   /// This method does not establish a relationship with the target in the way
-  /// that [inheritFromWidgetOfExactType] does.
+  /// that [dependOnInheritedWidgetOfExactType] does.
   ///
   /// This method should not be called from [State.dispose] because the element
   /// tree is no longer stable at that time. To refer to an ancestor from that
   /// method, save a reference to the ancestor by calling
-  /// [inheritFromWidgetOfExactType] in [State.didChangeDependencies]. It is
+  /// [dependOnInheritedWidgetOfExactType] in [State.didChangeDependencies]. It is
   /// safe to use this method from [State.deactivate], which is called whenever
   /// the widget is removed from the tree.
+  @Deprecated('Use ancestorElementForInheritedWidgetOfExactType instead')
   InheritedElement ancestorInheritedElementForWidgetOfExactType(Type targetType);
 
+  /// Obtains the element corresponding to the nearest widget of the given type [T],
+  /// which must be the type of a concrete [InheritedWidget] subclass.
+  ///
+  /// Calling this method is O(1) with a small constant factor.
+  ///
+  /// This method does not establish a relationship with the target in the way
+  /// that [dependOnInheritedWidgetOfExactType] does.
+  ///
+  /// This method should not be called from [State.dispose] because the element
+  /// tree is no longer stable at that time. To refer to an ancestor from that
+  /// method, save a reference to the ancestor by calling
+  /// [dependOnInheritedWidgetOfExactType] in [State.didChangeDependencies]. It is
+  /// safe to use this method from [State.deactivate], which is called whenever
+  /// the widget is removed from the tree.
+  InheritedElement ancestorElementForInheritedWidgetOfExactType<T extends InheritedWidget>();
```

Comment:
Name this getElementForInheritedWidgetOfExactType (I edited my suggestion shortly after posting, sorry if you only saw the old comment)?

Reply:
Renamed.